packages/flutter/lib/src/widgets/default_text_editing_actions.dart

```diff
@@ -0,0 +1,243 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+
+import 'actions.dart';
+import 'editable_text.dart';
+import 'framework.dart';
+import 'text_editing_action.dart';
+import 'text_editing_intents.dart';
+
+/// An [Actions] widget that handles the default text editing behavior for
+/// Flutter on the current platform.
+///
+/// This default behavior can be overridden by placing an [Actions] widget lower
+/// in the widget tree than this. See [DefaultTextEditingShortcuts] for an example of
+/// remapping keyboard keys to an existing text editing [Intent].
+///
+/// See also:
+///
+///   * [DefaultTextEditingShortcuts], which maps keyboard keys to many of the
+///     [Intent]s that are handled here.
+class DefaultTextEditingActions extends StatelessWidget {
+  /// Creates an instance of DefaultTextEditingActions.
+  const DefaultTextEditingActions({
+    Key? key,
+    required this.child,
+  }) : super(key: key);
+
+  /// The child [Widget] of DefaultTextEditingActions.
+  final Widget child;
+
+  static final TextEditingAction<ExpandSelectionLeftByLineTextIntent> _expandSelectionLeftByLineAction = TextEditingAction<ExpandSelectionLeftByLineTextIntent>(
```

Comment:
These are all private, so if I don't want to use all the defaults, I have to reimplement these?

Reply:
The main reason that I'm hesitant is just because I worry that I'll end up changing something with the API before I finish all of the other text editing actions.  How about if I leave them private for now and plan to make them public later when everything is finalized?

Reply:
Another thing I realized is that since I give the specific intent type for the type parameter (in this case ExpandSelectionLeftByLineTextIntent), these could not be used by any other Intent.  I could change the type parameter to just Intent to get around that.